### ISSUE TITLE
find_python() unable to find conda binary

### ISSUE THREAD

**Issue by marknovak** [dkoslicki/PressPurtCoreAlg#31]
Starting on a completely clean machine, installed miniconda, but find_python can't find conda binary.
~~~~~~~~~~
(base) Zugspitze:~ marknovak$ conda list
# packages in environment at /opt/miniconda3:
#
# Name                    Version                   Build  Channel
asn1crypto                1.0.1                    py37_0  
ca-certificates           2019.8.28                     0  
certifi                   2019.9.11                py37_0  
cffi                      1.12.3           py37hb5b8e2f_0  
chardet                   3.0.4                 py37_1003  
conda                     4.7.12                   py37_0  
conda-package-handling    1.6.0            py37h1de35cc_0  
cryptography              2.7              py37ha12b0ac_0  
idna                      2.8                      py37_0  
libcxx                    4.0.1                hcfea43d_1  
libcxxabi                 4.0.1                hcfea43d_1  
libedit                   3.1.20181209         hb402a30_0  
libffi                    3.2.1                h475c297_4  
ncurses                   6.1                  h0a44026_1  
openssl                   1.1.1d               h1de35cc_2  
pip                       19.2.3                   py37_0  
pycosat                   0.6.3            py37h1de35cc_0  
pycparser                 2.19                     py37_0  
pyopenssl                 19.0.0                   py37_0  
pysocks                   1.7.1                    py37_0  
python                    3.7.4                h359304d_1  
python.app                2                        py37_9  
readline                  7.0                  h1de35cc_5  
requests                  2.22.0                   py37_0  
ruamel_yaml               0.15.46          py37h1de35cc_0  
setuptools                41.4.0                   py37_0  
six                       1.12.0                   py37_0  
sqlite                    3.30.0               ha441bb4_0  
tk                        8.6.8                ha441bb4_0  
tqdm                      4.36.1                     py_0  
urllib3                   1.24.2                   py37_0  
wheel                     0.33.6                   py37_0  
xz                        5.2.4                h1de35cc_4  
yaml                      0.1.7                hc338f04_2  
zlib                      1.2.11               h1de35cc_3  
~~~~~~~~~~~~~~~
The above confirms that conda was successfully installed.
But, after opening up new R session (in R studio) and installing 'PressPurt', find_python() can't find the binary:
~~~~~~~~~~~~~
> # library('devtools')
> # devtools::install_github("dkoslicki/PressPurt", subdir = "R_version", build_vignettes = FALSE)
> library('PressPurt')
> find_python()
Error: Unable to find conda binary. Is Anaconda installed?
> find_python()
~~~~~~~~
Is the fact that miniconda was installed the problem?

**Comment by marknovak** [comment]
find_python() is not looking in all potential paths, including the default install location (which in my case was ~/usr/opt/...).

**Comment by marknovak** [comment]
I got things working on my laptop!  
The Miniconda I had previously installed did not work (unfortunately I didn’t write down what version that was.
So I tried an older version (4.6.14 from 2019/4/19), which worked!
So I tried the current version (4.7.12), which also worked!!

I also got it working on my desktop(!) with an older-than-previously-installed version of Anaconda (3.7.1.final.0 from 2018/12).
So then I tried the current version of Miniconda (4.7.12) which did NOT work !?!?!
So then I tried the current version of Anaconda which also did not work!?!?!

The error I’m getting for the last two is simply that find_python() can’t find the conda binary.  What’s interesting is the uninstalling with just “rm -rf ~/anaconda3” does not work!   I have to specify the full path “rm -rf /Users/marknovak/opt/anaconda3”.  That’s the same problem we had when you were in my office, Dana.  (i.e. the “opt” folder isn’t among the paths that find_python() is looking in.  Note that I just followed the defaults when installing.

So, then I uninstalled Anaconda and reinstalled the latest version of Miniconda.  However, having observed that on my laptop Miniconda is located under in the same location as ~/Users/ (i.e. in the master folder), I did not choose the default location but instead specified the installation path to be in the master folder.  Again, however, it placed the folder within an ‘opt’ folder, which is a hidden folder!  Interestingly, this ‘opt’ folder also had within it an ‘X11’ and a ‘local’ folder, the first of which I think might be associated with Xcode.  I therefore moved the ‘miniconda3’ out of ‘opt’ and into the master folder.  After that, everything works!

So yeah, the root cause of it all has been the default install location and that find_python() doesn’t look everywhere it needs to look.
